qdrant_client/common/retry.py

```
@@ -0,0 +1,405 @@
+
+
+@dataclass(frozen=True)
+class RetryConfig:
+    """User-facing retry configuration.
+
+    Args:
+        max_retries: Maximum number of retry attempts (0 = no retries). Default: 3.
+        initial_backoff: Base backoff duration in seconds. Default: 0.1.
+        max_backoff: Upper bound on backoff duration in seconds. Default: 30.
+        backoff_multiplier: Multiplier applied per attempt. Default: 2.
+        jitter: Jitter factor (0–1) added to backoff to avoid thundering herd. Default: 0.1.
```

Ruff (0.15.5)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)